diff/cert.go

```diff
@@ -85,6 +85,18 @@ func (sc *Syncer) createUpdateCertificate(
 
 	// found, check if update needed
 	if !currentCertificate.EqualWithOpts(certificateCopy, false, true) {
+		currentSNIs, err := sc.currentState.SNIs.GetAllByCertID(*currentCertificate.ID)
+		if err != nil {
+			return nil, errors.Wrapf(err, "error looking up current certificate SNIs %v",
+				certificate.Identifier())
+		}
+		sniNames := make([]*string, 0)
```

Comment:
```suggestion
		sniNames := make([]*string, 0, len(currentSNIs))
```
:point_up: just a micro-optimization: since we know the length this slice is going to end up needing to be anyway, this will heap alloc the right size array behind the slice instead of making the runtime do it dynamically.